src/app_service/service/message/service.nim

```diff
@@ -181,22 +166,34 @@ QtObject:
     # the same (once we may have replied messages before once after the messages one replied to), that's why we are
     # covering the most general case here.
     var messagesOneRepliedTo: seq[string]
-    for m in messages:
-      if m.responseTo.len > 0:
-        messagesOneRepliedTo.add(m.responseTo)
+
+    for msg in messages:
+      if(msg.editedAt > 0):
+        let data = MessageEditedArgs(chatId: msg.chatId, message: msg)
+        self.events.emit(SIGNAL_MESSAGE_EDITED, data)
+      if msg.responseTo.len > 0:
+        messagesOneRepliedTo.add(msg.responseTo)
 
     for msgId in messagesOneRepliedTo:
       removeMessageWithId(messages, msgId)
 
-    let data = MessagesArgs(
-      sectionId: sectionId,
-      chatId: chatId,
-      chatType: chatType,
-      unviewedMessagesCount: unviewedMessagesCount,
-      unviewedMentionsCount: unviewedMentionsCount,
-      messages: messages
-    )
-    self.events.emit(SIGNAL_NEW_MESSAGE_RECEIVED, data)
+    for i in 0 ..< chats.len:
+      if(chats[i].chatType == ChatType.Unknown):
+        error "error: new message with an unknown chat type received", chatId=chats[i].id
+        continue
+      
+      if chats[i].communityId.len == 0:
+        chats[i].communityId = singletonInstance.userProfile.getPubKey()
```

Comment:
looks super suspicious, why is that :thinking:?

wasn't the code below more descriptive/precise?
```
let sectionId = if chats[i].communityId.len > 0: chats[i].communityId else: singletonInstance.userProfile.getPubKey()
```

Reply:
@osmaczko what's the difference you see between this:
```
let sectionId = if chats[i].communityId.len > 0: chats[i].communityId else: singletonInstance.userProfile.getPubKey()

# and then in the place of usage we had this line:
sectionId: sectionId,
```
and this:
```
if chats[i].communityId.len == 0:
   chats[i].communityId = singletonInstance.userProfile.getPubKey()

# now in the place of usage we have this line:
sectionId: chats[i].communityId,
```

so in terms of code, we're doing the same thing, if `communityId` is not set, we set it to public key of the logged in user, but the difference is that we're sending signal for each chat now instead only for a single chat positioned on the index `0` in the `chats` array, and that was the issue actually.

Reply:
If this condition `chats[i].communityId.len == 0` is true, means that chat belongs to `Chat` section, and since we switched to channel group id of the `Chat` section is the same as public key of logged in user.